engine/src/main/java/org/terasology/engine/rendering/primitives/ChunkMesh.java

```diff
@@ -353,33 +304,48 @@ public int getTimeToGenerateOptimizedBuffers() {
      */
     public static class VertexElements {
 
-        public final TFloatList normals;
-        public final TFloatList vertices;
-        public final TFloatList tex;
-        public final TFloatList color;
-        public final TIntList indices;
-        public final TIntList flags;
-        public final TIntList frames;
-        public final TFloatList sunlight;
-        public final TFloatList blocklight;
-        public final TFloatList ambientOcclusion;
-        public int vertexCount;
-
-        public IntBuffer finalVertices;
-        public IntBuffer finalIndices;
+        public static final int VERTEX_INDEX = 0; // vec3
+        public static final int NORMAL_INDEX = 1;  // vec3
+        public static final int UV0_INDEX = 2;  // vec3
+
+        public static final int FLAGS_INDEX = 3;  // uint
+        public static final int FRAME_INDEX = 4; // uint
+
+        public static final int SUNLIGHT_INDEX = 5; // float
+        public static final int BLOCK_INDEX = 6; // float
+        public static final int AMBIENT_OCCLUSION_INDEX = 7; // float
+
+        public final VertexResource buffer;
+        public final IndexResource indices = new IndexResource();
+
+        public final VertexAttributeBinding<Vector3fc, Vector3f> position;
+        public final VertexAttributeBinding<Vector3fc, Vector3f> normals;
+        public final VertexAttributeBinding<Vector2fc, Vector2f> uv0;
+//        public final VertexAttributeBinding<Colorc, Color> color; // color data is unused
```

Comment:
remove?

Reply:
I want to keep a note here because we had color data that we're not using. I guess the plan was to have colored lights of some kind :?

Reply:
There were _beautiful_ colored torches long ago, in either case. This might have been left in place by somebody disabling them "temporarily" hoping they'd be back?